src/commands/contract/new.ts

```diff
@@ -0,0 +1,88 @@
+import { Command, Flags } from "@oclif/core";
+import path = require("node:path");
+import * as fs from 'fs-extra';
+import { getTemplates } from "../init";
+import { ensureSwankyProject } from "../../lib/command-utils";
+import { email, name, pickTemplate } from "../../lib/prompts";
+import { Spinner } from "../../lib/spinner";
+import {
+  checkCliDependencies,
+  copyContractTemplateFiles,
+  processTemplates,
+} from "../../lib/tasks";
+import { paramCase, pascalCase, snakeCase } from "change-case";
+import execa = require("execa");
+import inquirer = require("inquirer");
+
+export class NewContract extends Command {
+  static description = "Generate a new smart contract template inside a project";
+
+  static flags = {
+    template: Flags.string({
+      options: getTemplates().contractTemplatesList.map((template) => template.value),
+    }),
+    verbose: Flags.boolean({ char: "v" }),
+  }
+
+  static args = [
+    {
+      name: "contractName",
+      required: true,
+      description: "name of new contract"
+    }
+  ]
+
+  async run(): Promise<void> {
+    await ensureSwankyProject();
+
+    const projectPath = path.resolve();
+    const { args, flags } = await this.parse(NewContract);
+
+    if (fs.existsSync(path.join(projectPath, "contracts", args.contractName))) {
```

Comment:
better to use `pathExists` provided by fs-extra:
https://github.com/jprichardson/node-fs-extra/blob/master/docs/pathExists.md
(or sync version)